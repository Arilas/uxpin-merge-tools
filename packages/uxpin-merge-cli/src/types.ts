export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
