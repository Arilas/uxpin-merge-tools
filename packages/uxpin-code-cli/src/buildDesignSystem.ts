import * as webpack from 'webpack';

import { createComponentsLibrary } from './building/library/createComponentsLibrary';
import { BabelPlugin } from './building/plugins/BabelPluginDefinition';
import { getDesignSystemComponentInfos } from './components/getDesignSystemComponentInfos';
import { getConfig } from './config/webpack.config';

export function buildDesignSystem(babelPlugins:BabelPlugin[], wrapperPath?:string,
  target?:string):Promise<webpack.Stats> {
  return createLibrary(wrapperPath)
    .then(() => bundle(babelPlugins, target));
}

function bundle(babelPlugins:BabelPlugin[] = [], target?:string):Promise<webpack.Stats> {
  return new Promise((resolve, reject) => {
    const compiler:webpack.Compiler = webpack(getConfig(babelPlugins, target));

    compiler.run((err, stats) => {
      if (err) {
        return reject(err.message);
      }

      if (stats.hasErrors()) {
        return reject(stats.toString({ errors: true }));
      }

      resolve(stats);
    });
  });
}

function createLibrary(wrapperPath?:string):Promise<void> {
  return getDesignSystemComponentInfos()
    .then((componentInfos) => createComponentsLibrary(componentInfos, wrapperPath));
}
