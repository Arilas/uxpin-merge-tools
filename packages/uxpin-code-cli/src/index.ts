export { getDesignSystemMetadata } from './getDesignSystemMetadata';
export { getDesignSystemSummary } from './getDesignSystemSummary';
