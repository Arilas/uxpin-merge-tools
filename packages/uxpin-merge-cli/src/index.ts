import 'source-map-support/register';

export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';

export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
