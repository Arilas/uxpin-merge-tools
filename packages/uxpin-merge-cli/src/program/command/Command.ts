export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  CREATE_APP = 'create-app',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND:Command = Command.EXPERIMENT;
