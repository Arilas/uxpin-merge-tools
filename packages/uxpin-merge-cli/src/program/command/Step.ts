import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun:boolean;
  exec:(infos:DSMetadata) => any;
}
