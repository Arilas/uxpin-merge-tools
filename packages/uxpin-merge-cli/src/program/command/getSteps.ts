import { BuildOptions } from '../../steps/building/BuildOptions';
import { ProgramArgs, PushProgramArgs, ServerProgramArgs } from '../ProgramArgs';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args:ProgramArgs):Step[] {
  if (args.command === 'server') {
    return getServerCommandSteps(getBuildOptions(args), args);
  }

  if (args.command === 'push') {
    return getPushCommandSteps(getBuildOptions(args));
  }

  if (args.command === 'dump') {
    return getDumpCommandSteps();
  }

  if (args.command === 'summary') {
    return getSummaryCommandSteps();
  }

  return [];
}

function getBuildOptions(args:ServerProgramArgs | PushProgramArgs):BuildOptions {
  const { cwd, webpackConfig, wrapper } = args;
  return {
    projectRoot: cwd,
    webpackConfigPath: webpackConfig,
    wrapperPath: wrapper,
  };
}
