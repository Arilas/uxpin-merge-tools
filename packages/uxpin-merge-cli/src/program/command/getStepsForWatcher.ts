import { ProgramArgs } from '../args/ProgramArgs';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args:ProgramArgs):Step[] {
  if (args.command === 'experiment') {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
