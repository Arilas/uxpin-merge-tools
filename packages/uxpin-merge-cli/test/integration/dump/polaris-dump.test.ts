import { Command } from '../../../src';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';

const CURRENT_TIMEOUT:number = 120000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('The dump command', () => {
  it('run for the polaris repository, prints the JSON describing the full repository', () => {
    // when
    return runUXPinMergeCommand({
      cwd: 'resources/repos/polaris',
      params: [Command.DUMP, '--config="../../configs/polaris-uxpin.config.js"'],
    }).then((consoleOutput) => {
      // then
      expect(JSON.parse(consoleOutput)).toMatchSnapshot();
    });
  });
});
