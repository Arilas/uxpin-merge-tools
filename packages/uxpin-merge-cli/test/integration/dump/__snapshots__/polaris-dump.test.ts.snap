// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The dump command run for the polaris repository, prints the JSON describing the full repository 1`] = `
Object {
  "categorizedComponents": Array [
    Object {
      "components": Array [
        Object {
          "documentation": Object {
            "examples": Array [
              Object {
                "code": "<Avatar customer name=\\"Farrah\\" />",
              },
            ],
          },
          "info": Object {
            "dirPath": "src/components/Avatar",
            "documentation": Object {
              "path": "src/components/Avatar/README.md",
            },
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Avatar/Avatar.tsx",
            },
            "presets": Array [],
          },
          "name": "Avatar",
          "presets": Array [],
          "properties": Array [
            Object {
              "defaultValue": Object {
                "value": "medium",
              },
              "description": "",
              "isRequired": false,
              "name": "size",
              "type": Object {
                "name": "unsupported",
                "structure": Object {
                  "raw": "export type Size = 'small' | 'medium' | 'large' ;",
                },
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "name",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "initials",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "customer",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "source",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "accessibilityLabel",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [
              Object {
                "code": "<Button>Add product</Button>",
              },
              Object {
                "code": "<Button outline>Add product</Button>",
              },
              Object {
                "code": "<Button plain>View shipping settings</Button>",
              },
              Object {
                "code": "<Button primary>Save theme</Button>",
              },
              Object {
                "code": "<Button destructive>Delete theme</Button>",
              },
              Object {
                "code": "<Button size=\\"slim\\">Save variant</Button>",
              },
              Object {
                "code": "<Button size=\\"large\\">Create store</Button>",
              },
              Object {
                "code": "<Button fullWidth>Add customer</Button>",
              },
              Object {
                "code": "<Button disabled>Buy shipping label</Button>",
              },
              Object {
                "code": "<Button loading>Save product</Button>",
              },
            ],
          },
          "info": Object {
            "dirPath": "src/components/Button",
            "documentation": Object {
              "path": "src/components/Button/README.md",
            },
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Button/Button.tsx",
            },
            "presets": Array [],
          },
          "name": "Button",
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": false,
              "name": "url",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "children",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "size",
              "type": Object {
                "name": "unsupported",
                "structure": Object {
                  "raw": "export type Size = 'slim' | 'large';",
                },
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "fullWidth",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "primary",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "outline",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "destructive",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "disabled",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "loading",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "plain",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "external",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "submit",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "disclosure",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "accessibilityLabel",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "icon",
              "type": Object {
                "name": "unsupported",
                "structure": Object {
                  "raw": "IconProps['source']",
                },
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "onClick",
              "type": Object {
                "name": "func",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "onFocus",
              "type": Object {
                "name": "func",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "onBlur",
              "type": Object {
                "name": "func",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [
              Object {
                "code": "<Card title=\\"Online store dashboard\\" sectioned>
  <p>View a summary of your online store’s performance.</p>
</Card>",
              },
              Object {
                "code": "<Card
  title=\\"Online store dashboard\\"
  primaryFooterAction={{content: 'View dashboard'}}
>
  <Card.Section>
    <p>View a summary of your online store’s performance.</p>
  </Card.Section>
</Card>",
              },
              Object {
                "code": "<Card title=\\"Online store dashboard\\" actions={[{content: 'Edit'}]}>
  <Card.Section>
    <p>View a summary of your online store’s performance.</p>
  </Card.Section>
</Card>",
              },
              Object {
                "code": "<Card title=\\"Online store dashboard\\" actions={[{content: 'Edit'}]}>
  <Card.Section>
    <p>View a summary of your online store’s performance.</p>
  </Card.Section>
</Card>",
              },
              Object {
                "code": "<Card title=\\"Online store dashboard\\">
  <Card.Section>
    <p>View a summary of your online store’s performance.</p>
  </Card.Section>

  <Card.Section>
    <p>View a summary of your online store’s performance, including sales, visitors, top products, and referrals.</p>
  </Card.Section>
</Card>",
              },
              Object {
                "code": "<Card title=\\"Online store dashboard\\">
  <Card.Section title=\\"Reports\\">
    <p>View a summary of your online store’s performance.</p>
  </Card.Section>

  <Card.Section title=\\"Summary\\">
    <p>View a summary of your online store’s performance, including sales, visitors, top products, and referrals.</p>
  </Card.Section>
</Card>",
              },
              Object {
                "code": "<Card title=\\"Staff accounts\\">
  <Card.Section>
    <List>
      <List.Item>Felix Crafford</List.Item>
      <List.Item>Ezequiel Manno</List.Item>
    </List>
  </Card.Section>

  <Card.Section subdued title=\\"Deactivated staff accounts\\">
    <List>
      <List.Item>Felix Crafford</List.Item>
      <List.Item>Ezequiel Manno</List.Item>
    </List>
  </Card.Section>
</Card>",
              },
              Object {
                "code": "<Card title=\\"Deactivated staff accounts\\" sectioned subdued>
  <List>
    <List.Item>Felix Crafford</List.Item>
    <List.Item>Ezequiel Manno</List.Item>
  </List>
</Card>",
              },
            ],
          },
          "info": Object {
            "dirPath": "src/components/Card",
            "documentation": Object {
              "path": "src/components/Card/README.md",
            },
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/Card.tsx",
            },
            "presets": Array [],
          },
          "name": "Card",
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": false,
              "name": "title",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "subdued",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "sectioned",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "actions",
              "type": Object {
                "name": "array",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "primaryFooterAction",
              "type": Object {
                "name": "shape",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "secondaryFooterAction",
              "type": Object {
                "name": "shape",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Choice",
            "documentation": Object {
              "path": "src/components/Choice/README.md",
            },
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Choice/Choice.tsx",
            },
            "presets": Array [],
          },
          "name": "Choice",
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "id",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": true,
              "name": "label",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "error",
              "type": Object {
                "name": "unsupported",
                "structure": Object {
                  "raw": "export type Error = boolean | string;",
                },
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "labelHidden",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "helpText",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [
              Object {
                "code": "<Collapsible open>
  <TextContainer>
    Your mailing list lets you contact customers or visitors who have shown an interest in your store. Reach out to them with exclusive offers or updates about your products.
  </TextContainer>
</Collapsible>",
              },
            ],
          },
          "info": Object {
            "dirPath": "src/components/Collapsible",
            "documentation": Object {
              "path": "src/components/Collapsible/README.md",
            },
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Collapsible/Collapsible.tsx",
            },
            "presets": Array [],
          },
          "name": "Collapsible",
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "open",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [
              Object {
                "code": "<DescriptionList
  items={[
    {
      term: 'Logistics',
      description: 'The management of products or other resources as they travel between a point of origin and a destination.',
    },
    {
      term: 'Sole proprietorship',
      description: 'A business structure where a single individual both owns and runs the company.',
    },
    {
      term: 'Discount code',
      description: 'A series of numbers and/or letters that an online shopper may enter at checkout to get a discount or special offer.',
    },
  ]}
/>",
              },
            ],
          },
          "info": Object {
            "dirPath": "src/components/DescriptionList",
            "documentation": Object {
              "path": "src/components/DescriptionList/README.md",
            },
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/DescriptionList/DescriptionList.tsx",
            },
            "presets": Array [],
          },
          "name": "DescriptionList",
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "items",
              "type": Object {
                "name": "array",
                "structure": Object {},
              },
            },
          ],
        },
      ],
      "name": "Uncategorized",
    },
  ],
  "name": "",
}
`;
